auth/couchdb_session_authenticator.go

```diff
@@ -0,0 +1,225 @@
+// © Copyright IBM Corporation 2020. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/IBM/go-sdk-core/v4/core"
+)
+
+const (
+	AUTHTYPE_COUCHDB_SESSION = "couchdbSession"
+)
+
+var requestSessionMutex sync.Mutex
+
+// CouchDbSessionAuthenticator uses username and password to obtain
+// CouchDB authentication cookie, and adds the cookie to requests.
+type CouchDbSessionAuthenticator struct {
+	URL, Username, Password string
+	Client                  *http.Client
+	session                 *session
+}
+
+// GetAuthenticatorFromEnvironment instantiates an Authenticator
+// using service properties retrieved from external config sources.
+func GetAuthenticatorFromEnvironment(credentialKey string) (authenticator core.Authenticator, err error) {
+	props, err := core.GetServiceProperties(credentialKey)
+	if err != nil {
+		return
+	}
+	if props[core.PROPNAME_AUTH_TYPE] == "COUCHDB_SESSION" {
```

Comment:
The comparison is not case insensitive. We shall allow the `auth_type` is defined with lower case characters, too.

Reply:
Good catch! Addressed in https://github.com/IBM/cloudant-go-sdk/pull/15/commits/d610393a930b1f1abaf68e56a122db616b239a75